### PR TITLE
Added conversation search default use case with RAG tool

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Added conversation search default use case with RAG tool

### Issues Resolved
_List any issues this PR will resolve, e.g. Closes [...]._ 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by minalsha — APPROVED** [review]

**Comment by owaiskazi19** [comment]
Please merge this after @amitgalitz has taken a look.

File: src/main/resources/substitutionTemplates/conversational-search-with-bedrock-rag-tool-template.json [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+                            "access_key": "${{create_bedrock_connector.credential.access_key}}",
+                            "secret_key": "${{create_bedrock_connector.credential.secret_key}}",
+                            "session_token": "${{create_bedrock_connector.credential.session_token}}"
+                        },
+                        "parameters": {
+                            "max_tokens_to_sample": "8000",
+                            "service_name": "bedrock",
+                            "temperature": "1.0E-4",
+                            "response_filter": "$.completion",
+                            "region": "us-west-2",
```
Comment (amitgalitz) [inline src/main/resources/substitutionTemplates/conversational-search-with-bedrock-rag-tool-template.json:33]: user should have control over region

File: src/main/resources/substitutionTemplates/conversational-search-with-bedrock-rag-tool-template.json [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+                    },
+                    "user_inputs": {
+                        "type": "RAGTool",
+                        "name": "RAGTool",
+                        "parameters": {
+                            "inference_model_id": "${{register_bedrock_model.model_id}}",
+                            "embedding_model_id": "${{register_bedrock_embedding_model.model_id}}",
+                            "index": "${{create_index.name}}",
+                            "embedding_field": "${{text_embedding.field_map.output}}",
+                            "source_field": "[\"${{text_embedding.field_map.input}}\"]",
```
Comment (amitgalitz) [inline src/main/resources/substitutionTemplates/conversational-search-with-bedrock-rag-tool-template.json:185]: is this inside arrays on purpose

**Review by amitgalitz — APPROVED** [review]

File: src/main/resources/substitutionTemplates/conversational-search-with-bedrock-rag-tool-template.json (hunk 2 above)
Comment (owaiskazi19) [reply src/main/resources/substitutionTemplates/conversational-search-with-bedrock-rag-tool-template.json:185]: Yes, that's how ml-commons has it
```json
"source_field": [
        "passage_text"
      ]
```

**Pull request merged**